docs: add MongoDB Authenticated Charts integration guide (#3080)
caos/zitadel: docs/docs/guides/integrations/authenticated-mongodb-charts.md

---
title: Authenticated MongoDB Charts
---

This integration guide shows how you can embed authenticated MongoDB Charts in your web application using ZITADEL as authentication provider.

## Setup ZITADEL Application

Before you can embed an authenticated chart in your application, you have to do a few configuration steps in ZITADEL Console.
You will need to provide some information about your app. We recommend creating a new app to start from scratch.

1. Navigate to your [Project](https://console.zitadel.ch/projects)

> 1. Add a new application at the top of the page.

Reviewer: ```suggestion
2. Add a new application at the top of the page.
```

Author: in rendered markdown, numbers are incremented automatically, but sorting is easier when writing :-)

> 1. Select Web application type and continue.

Reviewer: ```suggestion
3. Select Web application type and continue.
```

> 1. Use [Authorization Code](../../apis/openidoauth/grant-types#authorization-code) in combination with [Proof Key for Code Exchange (PKCE)](../../apis/openidoauth/grant-types#proof-key-for-code-exchange).

Reviewer: ```suggestion
4. Use [Authorization Code](../../apis/openidoauth/grant-types#authorization-code) in combination with [Proof Key for Code Exchange (PKCE)](../../apis/openidoauth/grant-types#proof-key-for-code-exchange).
```

> 1. Skip the redirect settings and confirm the app creation

Reviewer: ```suggestion
5. Skip the redirect settings and confirm the app creation
```

> 1. Copy the client ID, you will need to tell MongoDB Charts about it.

Reviewer: ```suggestion
6. Copy the client ID, you will need to tell MongoDB Charts about it.
```

Reviewer 2: ... you will need to configure MongoDB Charts with it.

> 1. When you created the app, expand its _OIDC Configuration_ section, change the _Auth Token Type_ to _JWT_ and save the change.

Reviewer: ```suggestion
7. When you created the app, expand its _OIDC Configuration_ section, change the _Auth Token Type_ to _JWT_ and save the change.
```

Your application configuration should now look similar to this:

![Create app in console](/img/integrations/mongodb-charts-app-create-light.png)

## Setup Custom JWT Provider for MongoDB Charts

Configure ZITADEL as your _Custom JWT Provider_ following the [MongoDB docs](https://docs.mongodb.com/charts/configure-auth-providers/) .

Configure the following values:
- Signing Algorithm: RS256
- Signing Key: JWK or JWKS URL
- JWKS: https://api.zitadel.ch/oauth/v2/keys

> - Audience: Your apps client ID, you copied when you created the ZITADEL app

Reviewer 2: ```suggestion
- Audience: Your app's client ID which you copied when you created the ZITADEL app
```
